### PR TITLE
Test that defaults in stub files applied to classes are used

### PR DESCRIPTION
**Pull request by zcai1**
This PR aims to answer the question raised in https://github.com/eisop/checker-framework/pull/269#discussion_r911459244

We always trust the given astub files. This PR adds a test to ensure defaults on a class will not affect its child classes.

### CODE REVIEW

File: checker/jtreg/stubs/inheritance-in-stub/pet.astub [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class Pet {
```
Comment (zcai1) [inline checker/jtreg/stubs/inheritance-in-stub/pet.astub:3]: default qualifiers are only applied to the class

File: checker/jtreg/stubs/defaultqualinstub/default-on-class/PetTest.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * @compile -XDrawDiagnostics -Xlint:unchecked Pet.java
+ * @compile -XDrawDiagnostics -Xlint:unchecked Cat.java
+ * @compile/fail/ref=PetTest.out -XDrawDiagnostics -Xlint:unchecked -processor org.checkerframework.checker.nullness.NullnessChecker -Anomsgtext -Astubs=pet.astub -Werror PetTest.java
+ */
+
+public class PetTest {
+    void foo() {
+        Cat c = new Cat();
+        c.playWith(null);
```
Comment (zcai1) [inline checker/jtreg/stubs/defaultqualinstub/default-on-class/PetTest.java:13]: add test for pet

**Review by wmdietl — COMMENTED** [review]
To be closer to the original problem in #269, could you try whether this correctly works with a wildcard in the stub file for Pet?
The default qualifier in a package-info also worked in general, it just didn't work for wildcards.

**Comment by zcai1** [comment]
> To be closer to the original problem in #269, could you try whether this correctly works with a wildcard in the stub file for Pet? The default qualifier in a package-info also worked in general, it just didn't work for wildcards.

for sure, I have updated the test case

**Comment by zcai1** [comment]
subset of https://github.com/eisop/checker-framework/pull/305

**Pull request closed without merging**